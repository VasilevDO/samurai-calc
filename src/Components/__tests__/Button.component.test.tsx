import {cleanup, fireEvent, render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Button from '../Button.component';

afterEach(cleanup);

describe('ButtonComponent', () => {
	const buttonRole = 'button';
	const buttonText = 'text';
	const buttonAction = jest.fn();
	const buttonStyle = '';

	test('should render button with correct text (no style provided)', () => {
		const {getByRole} = render(<Button text={buttonText} action={buttonAction}/>);
		const element = getByRole(buttonRole);
		expect(element).toBeInTheDocument();
		expect(element).toHaveTextContent(buttonText);
	});
	test('should render button with correct text (style provided)', () => {
		const {getByRole} = render(<Button text={buttonText} action={buttonAction} style={buttonStyle}/>);
		const element = getByRole(buttonRole);
		expect(element).toBeInTheDocument();
		expect(element).toHaveTextContent(buttonText);
	});
	test('should call props action function with text on click', () => {
		const {getByRole} = render(<Button text={buttonText} action={buttonAction}/>);
		const element = getByRole(buttonRole);
		fireEvent.click(element);
		expect(element).toBeInTheDocument();
		expect(buttonAction).toHaveBeenCalledWith(buttonText);
	});
});

