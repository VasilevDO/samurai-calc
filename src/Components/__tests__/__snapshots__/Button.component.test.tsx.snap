// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent tests snap 1`] = `
<button
  className="sc-bczRLJ jqbBEv"
  onClick={[Function]}
>
  kek
</button>
`;
